src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,98 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
+import { pattern } from '../Service/RegularFunction';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+const initialMovieState: Movie = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [movie, setMovie] = useState<Movie>(initialMovieState);
+  const handleMovieChange = (event: React.ChangeEvent<HTMLInputElement>,
+    key: string) => {
+    setMovie(prevMovie => ({ ...prevMovie, [key]: event.target.value }));
+  };
+
+  const isFormValid = (newMovie: Movie): boolean => {
+    if (!newMovie.title.trim() || !newMovie.imgUrl.trim()
+    || !newMovie.imdbId.trim() || !newMovie.imdbUrl.trim()) {
+      return false;
+    }
+
+    if ((!newMovie.imgUrl.match(pattern) && newMovie.imgUrl !== '')
+    || (!newMovie.imdbUrl.match(pattern) && newMovie.imdbUrl !== '')) {
+      return false;
+    }
+
+    return true;
```

Comment:
That's what we're talking about. Nice job, since we can even simplify this condition
```suggestion
   const { title, imgUrl, imdbId, imdbUrl } = newMovie;
  
  return (
    title.trim() &&
    imgUrl.trim() &&
    imdbId.trim() &&
    imdbUrl.trim() &&
    imgUrl.match(pattern) &&
    imdbUrl.match(pattern)
  );
```